DToCToDB.md

```diff
@@ -0,0 +1,1193 @@
+    </tr>
+    <tr>
+      <th>18</th>
+      <td>2016-17</td>
+      <td>OCTOBER</td>
+      <td>Q82</td>
+      <td>NHS ENGLAND SOUTH (SOUTH CENTRAL)</td>
+      <td>AXG</td>
+      <td>WILTSHIRE HEALTH &amp; CARE LLP</td>
+      <td>817</td>
+      <td>WILTSHIRE</td>
+      <td>Acute</td>
+      <td>B_PUBLIC_FUNDING</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>19</th>
+      <td>2016-17</td>
+      <td>OCTOBER</td>
+      <td>Q82</td>
+      <td>NHS ENGLAND SOUTH (SOUTH CENTRAL)</td>
+      <td>AXG</td>
+      <td>WILTSHIRE HEALTH &amp; CARE LLP</td>
+      <td>817</td>
+      <td>WILTSHIRE</td>
+      <td>Acute</td>
+      <td>A_COMPLETION_ASSESSMENT</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+The aim of this exercise is to combine all the available datasets into a single dataset, to then insert into a SQLite table within a database that contains other NHS datasets.
+
+To begin, we import all the datasets for 2016 into pandas dataframes using the `pd.read_csv` function. I have done this through a direct link to the files on the NHS England wesbite for the purposes of reproducability; note the use of the parameter `header=3` - this has been used as the csv header begins further down the page. 
+
+
+```python
+sep_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-hfU0D.csv', header=3)
+aug_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-dj498.csv', header=3)
+jul_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-9H0jr.csv', header=3)
+jun_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-T25Kd.csv', header=3)
+may_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-kYNdw.csv', header=3)
+apr_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2016/06/Monthly-SITREPs-DTOC-Extracts-nN06x.csv', header=3)
+mar_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-iFYJ2.csv', header=3)
+feb_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-gVela.csv', header=3)
+jan_16 = pd.read_csv('https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2015/05/Monthly-SITREPs-DTOC-Extracts-9jMj2.csv', header=3)
+```
+
+We then combine the 2016 dataframes in a single dataframe assigned to `dtoc_16` using the `pd.concat()` function:
+
+
+```python
+frames = [oct_16, sep_16, jul_16, jun_16, may_16, apr_16, mar_16, feb_16, jan_16]
```

Comment:
It's useful to do some validation before combining to ensure they have the same number of columns/column names.  you can check with a for loop to make sure the column names in each frame are the same.